bitnami/postgresql-ha/templates/postgresql/secrets.yaml

```
@@ -7,6 +7,9 @@ metadata:
     app.kubernetes.io/component: postgresql
 type: Opaque
 data:
+  {{- if and .Values.postgresql.postgresPassword (not (eq .Values.postgresql.username "postgres")) }}
```

I've carried these conditionals over from https://github.com/bitnami/charts/blob/1765929fa8cf37819309ff975ebf0a8dfcf7824d/bitnami/postgresql/templates/secrets.yaml#L16, but they seem weird. If you set `username` or `postgresPassword` via the `global` section this won't trigger.

This seems better implemented as
```suggestion
  {{- if and (include "postgresql-ha.postgresqlPostgresPassword" .) (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) }}
```

However the template provides a random default if you don't specify one. Switching to checking the template result means that the `postgres` user would always have a password. If someone's intention was to disable that user they'd no longer have a way to do so.

---

Honestly, I think your solutions is much better.

> If someone's intention was to disable that user they'd no longer have a way to do so

That's not a problem, since it's required at this very moment due to the validations below:

- https://github.com/bitnami/bitnami-docker-postgresql-repmgr/blob/master/11/debian-10/rootfs/opt/bitnami/scripts/libpostgresql.sh#L247